mewarpx/mewarpx/mwxrun.py

```diff
@@ -103,23 +104,50 @@ def init_timestep(self, V_anode, DT=None, CFL_factor=None, V_grid=5):
         return self.dt
 
 
-    def init_run(self):
+    def init_run(self, restart=None, checkpoint_dir="diags",
+                 checkpoint_prefix="chkpoint", additional_steps=None):
         if self.initialized:
             raise RuntimeError(
                 "Attempted to initialize the mwxrun class multiple times.")
-
-        self.simulation.initialize_inputs()
-        self.simulation.initialize_warpx()
-
-        self.me = _libwarpx.libwarpx.warpx_getMyProc()
-        self.n_procs = _libwarpx.libwarpx.warpx_getNProcs()
-
-        # A level is needed for many things like level number. For now I'm
-        # statically setting the default level here. I'm not sure of pitfalls
-        # or how to handle it more generally yet.
-        self.lev = _libwarpx.libwarpx.warpx_finestLevel()
-
-        self.initialized = True
+        try:
+            restart_success = None
+            if restart != False:
+                restart_success = init_restart_util.run_restart(
+                    checkpoint_dir, checkpoint_prefix,
+                    restart is not None, additional_steps
+                )
+
+            self.simulation.initialize_inputs()
+            self.simulation.initialize_warpx()
+
+            self.me = _libwarpx.libwarpx.warpx_getMyProc()
+            self.n_procs = _libwarpx.libwarpx.warpx_getNProcs()
+
+            # A level is needed for many things like level number. For now I'm
+            # statically setting the default level here. I'm not sure of pitfalls
+            # or how to handle it more generally yet.
+            self.lev = _libwarpx.libwarpx.warpx_finestLevel()
+
+            self.initialized = True
+
+        except RuntimeError:
+            if restart != False:
+                if restart_success:
+                    logger.errro(
+                        "init_restart_util returned success for restarting from"
+                        f"a checkpoint starting with {checkpoint_prefix} "
+                        "but there was an error initializing WarpX!"
+                    )
+                else:
+                    logger.error(
+                        "There was an error restarting from "
+                        f"a checkpoint starting with {checkpoint_prefix}!"
+                    )
```

Comment:
This will be triggered if `restart_success = False` (for example there are no checkpoint files available), in which case the appropriate error message would be: `"There was an error initializing the run!"`. So I would just use that same error message here (which also means that you don't really need `restart` and `restart_success`.